test/engine/engine.js

```diff
@@ -357,9 +357,16 @@ for (const fixture of fixtures) {
             address.publicAddress
           }?api_key=MY_APIKEY`,
           (err, res, body) => {
-            const thirdPartyBalance = parseInt(JSON.parse(body).received)
-            assert(!err, 'getting address incoming txs from a second source')
-            assert(thirdPartyBalance === 0, 'Should have never received coins')
+            const code = parseInt(JSON.parse(body).code)
+            if (!err && code !== 429) {
+              const thirdPartyBalance = parseInt(JSON.parse(body).received)
+              assert(!err, 'getting address incoming txs from a second source')
+              assert(thirdPartyBalance === 0, 'Should have never received coins')
+            } else {
+              const scriptHash = engine.engineState.scriptHashes[address.publicAddress]
```

Comment:
I don't understand this. Seems like you are just letting the test pass without looking at the result from the API.

Reply:
Maybe these tests should really not depend on network calls at all and have that as a separate test that isn't auto run

Reply:
So that's why I first try to use the network call and if that fails, then I test for internal consistency of the engine since It's the best I can do without the 3rd party response.